twoliter/src/lock.rs

```diff
@@ -139,6 +183,121 @@ impl Lock {
         Ok(lock)
     }
 
+    fn external_kit_metadata(&self) -> ExternalKitMetadata {
+        ExternalKitMetadata {
+            sdk: self.sdk.clone(),
+            kits: self.kit.clone(),
+        }
+    }
```

Comment:
ExternalKitMetadata is a subset of the fields in the lock file, otherwise irrelevant information to cargo rerun would end up in the json

Reply:
> ExternalKitMetadata is a subset of the fields in the lock file, otherwise irrelevant information to cargo rerun would end up in the json

Oh, I see - it's because we're in `impl Lock` here. It'd be better not to have so many intervening data structures with their own impls between where `struct Lock` is defined and this point.

Reply:
Reordered some things to make this cleaner